src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,689 +1,906 @@
-            except subprocess.CalledProcessError as cpe:
-                raise RuntimeError(
-                    '\nNecessary file could not be acquired: %s. Got error "%s". Check input URL' % (url, cpe))
-            except OSError:
-                raise RuntimeError('Failed to find "curl". Install via "apt-get install curl"')
-    assert os.path.exists(file_path)
-    return job.fileStore.writeGlobalFile(file_path)
-
-
-def return_input_paths(job, work_dir, ids, *filenames):
-    """
-    Given one or more filenames, reads the file from the fileStore and
-    copies it (or makes a hard link) into the working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory, str
-    :param ids: dictionary of shared file ids
-    :param filenames: remaining arguments are filenames
-
-    :returns: list of paths to files
-    """
-    paths = OrderedDict()
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            file_path = job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-        else:
-            file_path = os.path.join(work_dir, filename)
-        paths[filename] = file_path
-
-    return paths.values()
-
-
-def write_to_filestore(job, work_dir, ids, *filenames):
-    """
-    Given one or more file names in working directory, writes
-    files to filestore and stores the filestore promise in a dictionary
+from toil_lib.urls import download_url_job
+from toil_lib.files import generate_file
+from toil_lib.tools.aligners import run_bwakit
+from toil_lib.tools.indexing import run_samtools_faidx
+from toil_lib.tools.preprocessing import run_gatk_preprocessing, \
+    run_picard_create_sequence_dictionary, run_samtools_index, run_samtools_sort
+from toil_lib.tools.variant_annotation import gatk_genotype_gvcfs, run_oncotator
+import yaml
 
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are keys for ids
+from toil_scripts.gatk_germline.common import output_file_job
+from toil_scripts.gatk_germline.germline_config import generate_config, generate_manifest
+from toil_scripts.gatk_germline.hard_filter import hard_filter_pipeline
+from toil_scripts.gatk_germline.vqsr import vqsr_pipeline
 
-    :returns: shared ids dictionary
-    """
-    for filename in filenames:
-        ids[filename] = job.fileStore.writeGlobalFile(os.path.join(work_dir, filename))
-    return ids
 
+logging.basicConfig(level=logging.INFO)
 
-def read_from_filestore_hc(job, work_dir, ids, *filenames):
-    """
-    Reads file from fileStore and writes it to working directory.
 
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are filenames
+class GermlineSample(namedtuple('GermlineSample', 'uuid url paired_url rg_line')):
```

Comment:
I think this is clearer because you can link to this class in docstrings and it documents the requirements for a sample in one location.